docs/useful-snippets/snippets.rst

```diff
@@ -58,3 +58,50 @@ code as follows.
 If you would still like to run your code in Jupyter you will need to adjust the cell as noted above. Then you can
 you can add the `nbmultitask library <(https://nbviewer.org/github/micahscopes/nbmultitask/blob/39b6f31b047e8a51a0fcb5c93ae4572684f877ce/examples.ipynb)>`__
 or look at this `stackoverflow <https://stackoverflow.com/questions/50937362/multiprocessing-on-python-3-jupyter>`__.
+
+Using multiple schedules and data collectors
+-------------
+
+One may occasionally want multiple schedules with distinct sets of agents and collect data on each of those schedules independently while allowing the agents to interact in the same model. This would be particularly useful if the agents in your model do not all have the same attributes and you want to collect data on them in groups. You can create a simple scheduler class to mix the agents in each schedule and data collect on each schedule independently.
+
+.. code:: python
+
+  from random import shuffle
+
+  # simple schedule mixer
+  class ScheduleMixerRandom:
+      def __init__(self, schedules: list):
+          self.schedules = schedules
+
+      def step(self):
+          agents = []
+          for s in self.schedules:
+              agents.extend(s.agents)
+          agents = shuffle(agents) # step through them randomly
```

Comment:
I think one of the main reason one wants to have multiple schedulers is so that agents in each of the schedulers are executed separately, in groups. It would have driven the point better with
```python
for s in self.schedules:
    s.step()
```

Reply:
Oh, nvm, the name of the class itself says that it is a mixer.

That makes me wonder. For the use case for when you just need a `RandomActivation` for the model step, but want to do something complex with data collection, it is rather overkill to have multiple schedulers just for slicing the agents list. While it's not used in model step at all. Maybe it is cleaner after all, with just passing an agents list to the `DataCollector` initialization, as @Corvince said in https://github.com/projectmesa/mesa/pull/1481#issuecomment-1297470299?

Reply:
In any case, the example needs to encapsulate properly the use case that is covered by this new feature. Reconstructing the agents list from scratch for every model step seem slow to me.

Reply:
For this snippet, I would define `ScheduleMixerRandom` as
```python
def __init__(self, schedules):
    self._agents = []
    for s in schedules:
        self._agents.extends(s.agents)

def step(self):
    # do stuff with self._agents
    ...
```

Reply:
> Maybe it is cleaner after all, with just passing an agents list to the DataCollector initialization, as @Corvince said in https://github.com/projectmesa/mesa/pull/1481#issuecomment-1297470299?

My off-the-cuff concern here would be if one wants to add/remove agents during the execution of the model. It would seem that if we passed a list at creation, we'd have to update both the schedule and the list in the `DataCollector` which feels sort of redundant and cumbersome. Also if we remove agents arbitrarily, we'd have to search the list in the `DataCollector` and find the right one to remove.

> In any case, the example needs to encapsulate properly the use case that is covered by this new feature. Reconstructing the agents list from scratch for every model step seem slow to me.

Yes, I wouldn't re-create the list each time in a real model, I'd probably do some sort of master list indexing or force interaction with the sub-schedules via the schedule mixer to ensure things update when necessary. My thought with the snippet was to show how this functionality is used in the dumbest possible way so there is no fancy coding or confusion as to what it is or how it works. Any optimization (especially of a custom class) or more elaborate uses would be the responsibility of the user/implementer and would not necessarily appear in a snippet.

Reply:
Yeah, the add/remove agent would have to be done redundantly, or it depends on the model. Though, the list search problem can be fixed if we use a dict.

There is a case if the main scheduler used by the model step is `RandomActivationByType`, which is a dict of dict. If the `DataCollector` accepts a custom dict, then you could pass in `model.schedule[agent_type]`. In this case, add/remove needs to be done once per agent death/birth event, since there is only 1 scheduler object.

> I wrote it the way I did to ensure the capture of any changes to the agent list like additions or subtractions.

OK, I can see how the implementation seeks to minimize error for the users who use the snippet. Ensuring correctness is indeed safer than performance. But at the same time, we could add 1 documenting sentence in `ScheduleMixerRandom` saying that the user needs to be mindful when adding/remove an agent.

Reply:
Yeah, it would be a bit ugly in any way.

Updated. Let me know if there are any additional concerns, thoughts, etc.